Delivered-To: w@wdtz.org
Received: by 2002:a4a:ab04:0:0:0:0:0 with SMTP id i4csp3401826oon;
        Mon, 15 Apr 2019 18:46:45 -0700 (PDT)
X-Google-Smtp-Source: APXvYqwqI+IzQSPkhAE9UGC29rsHvFEO9UZVl6i9xvxkT1KJtpRmMqaOg43WtwkFaayjaoCDKD91
X-Received: by 2002:a63:720e:: with SMTP id n14mr72789058pgc.93.1555379204972;
        Mon, 15 Apr 2019 18:46:44 -0700 (PDT)
ARC-Seal: i=1; a=rsa-sha256; t=1555379204; cv=none;
        d=google.com; s=arc-20160816;
        b=LORkd9UNSRSYu2v/O7wxlH8VvVBV6lhr2HJbDUXjW+lU0afKRUabcYpsKdzGw7WHvt
         yFkiaUb3AexcNvy64mHKSOxiAaK4M7/d2tfcP7G29iLargFDNFDhDIyCohy8haI2oyZU
         B8YucYqHKbAu7oP9eO3o30oLO8/TR/k/0lN6FXS0y0XKvXvoe1rQ47gdECsSbO/dDPmf
         GAndF0EnQ4MDr5JDqujPFVS3yKsCeS81ndG8k6TP+SXvAptGuenxtmJO5RbtT6zGhpEu
         S7u836+065OcOKGEpUDPLjIP0C2wY7h0/o6G+jqNRSOPClEsDUCfZWyb2xGo2DpnTAZE
         lcDg==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=google.com; s=arc-20160816;
        h=sender:errors-to:content-transfer-encoding:list-subscribe:list-help
         :list-post:list-archive:list-unsubscribe:list-id:precedence
         :mime-version:references:in-reply-to:message-id:date:subject:to:from
         :delivered-to;
        bh=Wn6tsk7Ksihi04VJQe80GZQuQcbZfVGEAgKjsColylU=;
        b=fn8lMGl5uMD2H05fKLLY5ryPFqSZqUGcGv4pby4qQIf1o9QWbUAjCKLaONNZIhkq7T
         q2L6Oyr3OD7bP6DOXBcCcEuE1F9gCujq/w+m7oao1/D2CSIC14XSZMxY3CaoyTbFVKyW
         KLk1Vv8LfyZvt6FGds3DVoE/7sWUEgVfAK+CK64Se5NLT4JeZCCJ5E63bHiImhETcXl8
         mbCNCWW1xE610BbZ+dLk/etSzvOgb79qQTCT0F4yQT+Dyw1E+TUNDEd1SEH/JUmkWN/T
         KMXvrrSQZmGyPpxQ/A6cKpfBzSSYwf1QV9jyjWHxD1GHFydqaQLZqe3gTqGGaglBTHoM
         ZwcA==
ARC-Authentication-Results: i=1; mx.google.com;
       spf=neutral (google.com: 50.126.95.6 is neither permitted nor denied by best guess record for domain of notmuch-bounces@notmuchmail.org) smtp.mailfrom=notmuch-bounces@notmuchmail.org
Return-Path: <notmuch-bounces@notmuchmail.org>
Received: from arlo.cworth.org (arlo.cworth.org. [50.126.95.6])
        by mx.google.com with ESMTPS id e4si46744770pgn.237.2019.04.15.18.46.44
        for <w@wdtz.org>
        (version=TLS1_2 cipher=ECDHE-RSA-CHACHA20-POLY1305 bits=256/256);
        Mon, 15 Apr 2019 18:46:44 -0700 (PDT)
Received-SPF: neutral (google.com: 50.126.95.6 is neither permitted nor denied by best guess record for domain of notmuch-bounces@notmuchmail.org) client-ip=50.126.95.6;
Authentication-Results: mx.google.com;
       spf=neutral (google.com: 50.126.95.6 is neither permitted nor denied by best guess record for domain of notmuch-bounces@notmuchmail.org) smtp.mailfrom=notmuch-bounces@notmuchmail.org
Received: from localhost (localhost [127.0.0.1])
	by arlo.cworth.org (Postfix) with ESMTP id 823736DE146A;
	Mon, 15 Apr 2019 18:46:38 -0700 (PDT)
X-Virus-Scanned: Debian amavisd-new at cworth.org
Received: from arlo.cworth.org ([127.0.0.1])
	by localhost (arlo.cworth.org [127.0.0.1]) (amavisd-new, port 10024)
	with ESMTP id KODqSdwCQXtu; Mon, 15 Apr 2019 18:46:38 -0700 (PDT)
Received: from arlo.cworth.org (localhost [IPv6:::1])
	by arlo.cworth.org (Postfix) with ESMTP id A44256DE1452;
	Mon, 15 Apr 2019 18:46:30 -0700 (PDT)
X-Original-To: notmuch@notmuchmail.org
Delivered-To: notmuch@notmuchmail.org
Received: from localhost (localhost [127.0.0.1])
 by arlo.cworth.org (Postfix) with ESMTP id 77FD06DE0F40
 for <notmuch@notmuchmail.org>; Mon, 15 Apr 2019 18:46:28 -0700 (PDT)
X-Virus-Scanned: Debian amavisd-new at cworth.org
Received: from arlo.cworth.org ([127.0.0.1])
 by localhost (arlo.cworth.org [127.0.0.1]) (amavisd-new, port 10024)
 with ESMTP id nej0tmmT0CN7 for <notmuch@notmuchmail.org>;
 Mon, 15 Apr 2019 18:46:28 -0700 (PDT)
Received: from fethera.tethera.net (fethera.tethera.net [198.245.60.197])
 by arlo.cworth.org (Postfix) with ESMTPS id 925726DE0F5A
 for <notmuch@notmuchmail.org>; Mon, 15 Apr 2019 18:46:27 -0700 (PDT)
Received: from remotemail by fethera.tethera.net with local (Exim 4.89)
 (envelope-from <bremner@tethera.net>)
 id 1hGDB4-0007pD-Ob; Mon, 15 Apr 2019 21:46:26 -0400
Received: (nullmailer pid 31746 invoked by uid 1000);
 Tue, 16 Apr 2019 01:46:21 -0000
From: David Bremner <david@tethera.net>
To: notmuch@notmuchmail.org
Subject: [PATCH 1/2] CLI/reindex: fix memory leak
Date: Mon, 15 Apr 2019 22:46:15 -0300
Message-Id: <20190416014616.31623-2-david@tethera.net>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190416014616.31623-1-david@tethera.net>
References: <20190416014616.31623-1-david@tethera.net>
MIME-Version: 1.0
X-BeenThere: notmuch@notmuchmail.org
X-Mailman-Version: 2.1.29
Precedence: list
List-Id: "Use and development of the notmuch mail system."
 <notmuch.notmuchmail.org>
List-Unsubscribe: <https://notmuchmail.org/mailman/options/notmuch>,
 <mailto:notmuch-request@notmuchmail.org?subject=unsubscribe>
List-Archive: <http://notmuchmail.org/pipermail/notmuch/>
List-Post: <mailto:notmuch@notmuchmail.org>
List-Help: <mailto:notmuch-request@notmuchmail.org?subject=help>
List-Subscribe: <https://notmuchmail.org/mailman/listinfo/notmuch>,
 <mailto:notmuch-request@notmuchmail.org?subject=subscribe>
Content-Type: text/plain; charset="us-ascii"
Content-Transfer-Encoding: 7bit
Errors-To: notmuch-bounces@notmuchmail.org
Sender: "notmuch" <notmuch-bounces@notmuchmail.org>

Since message is owned by messages, it was held for the entire run of
the program. This in turn means that the Xapian::Document objects are
not freed, and thus one ends up with (effectively) a copy of one's
entire mailstore in memory when running

       notmuch reindex '*'

Thanks to Olly Betts for the patient help debugging, and the
suggestion of a fix.
---
 notmuch-reindex.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/notmuch-reindex.c b/notmuch-reindex.c
index d8589120..fefa7c63 100644
--- a/notmuch-reindex.c
+++ b/notmuch-reindex.c
@@ -71,6 +71,7 @@ reindex_query (notmuch_database_t *notmuch, const char *query_string,
 	ret = notmuch_message_reindex(message, indexopts);
 	if (ret != NOTMUCH_STATUS_SUCCESS)
 	    break;
+	notmuch_message_destroy (message);
     }
 
     if (!ret)
-- 
2.20.1

_______________________________________________
notmuch mailing list
notmuch@notmuchmail.org
https://notmuchmail.org/mailman/listinfo/notmuch
